core-api/src/test/java/com/optimizely/ab/OptimizelyBuilderTest.java

```diff
@@ -195,4 +203,33 @@ public void withDefaultDecideOptions() throws Exception {
         assertEquals(optimizelyClient.defaultDecideOptions.get(2), OptimizelyDecideOption.EXCLUDE_VARIABLES);
     }
 
+    @Test
+    public void withClientInfo() throws Exception {
+        Optimizely optimizely;
+        EventHandler eventHandler;
+        ArgumentCaptor<LogEvent> argument = ArgumentCaptor.forClass(LogEvent.class);
+
+        // default client-engine info (java-sdk)
+
+        eventHandler = mock(EventHandler.class);
+        optimizely = Optimizely.builder(validConfigJsonV4(), eventHandler).build();
+        optimizely.track("basic_event", "tester");
+
+        verify(eventHandler, timeout(5000)).dispatchEvent(argument.capture());
+        assertEquals(argument.getValue().getEventBatch().getClientName(), "java-sdk");
+        assertEquals(argument.getValue().getEventBatch().getClientVersion(), BuildVersionInfo.getClientVersion());
+
+        // override client-engine info
+
+        eventHandler = mock(EventHandler.class);
+        optimizely = Optimizely.builder(validConfigJsonV4(), eventHandler)
+            .withClientInfo(EventBatch.ClientEngine.ANDROID_SDK, "1.2.3")
```

Comment:
isn't it confusing to check "android-sdk" client engine.

Reply:
Can you clarify this comment? Not clear to me.